packages/base/card-api.gts

```diff
@@ -3117,10 +3118,10 @@ export class Box<T> {
       );
     }
 
-    let { prevChildren, state } = this;
+    let { prevChildren, prevValues, state } = this;
     let newChildren: Box<ElementType<T>>[] = value.map((element, index) => {
-      let found = prevChildren.find((oldBox, i) =>
-        state.useIndexBasedKeys ? index === i : oldBox.value === element,
```

Comment:
The main issue was here. 

We were relying on getting  .value although that was already set during the array assigment. This caused us to always be checking arrays that was already of the same order